cli/tools/test.rs

```diff
@@ -1355,12 +1354,14 @@ pub async fn run_tests_with_watch(
   let no_check = ps.options.type_check_mode() == TypeCheckMode::None;
   let test_options = &test_options;
 
+  let ps = RefCell::new(ps);
```

Comment:
Now that I'm seeing it I'm a bit worried this might cause unexpected bugs - since `ProcState` is internally an `Arc` a lot of APIs could have cloned it and hold onto a copy of that `Arc`. Even if we create a new `Arc` in `reset_for_file_watcher` some code might hold onto the old copy. Does that make sense? Are am I overanalyzing it and essentially anything that can hold onto `ProcState` gets torn down between restarts?

Reply:
Well `ProcState`'s fields are either (1) immutable config, (2) mutable data that gets defaulted on reset, or (3) interior mutable fields with their own `Arc`s which are preserving-ly cloned on reset. If there _were_ an old lingering `ProcState` inner, the issue would be if (2) and (3) started conflicting since (3) is being affected by a future run. That can be avoided by being strict about what fields are (3), if there's an bug it should just mean that more fields need to be defaulted on reset.. I think

Reply:
Okay, makes sense. Thanks for investigating